src/uct/self/self_ep.h

```diff
@@ -0,0 +1,23 @@
+/**
+* Copyright (C) UT-Battelle, LLC. 2015. ALL RIGHTS RESERVED.
+* Copyright (C) Mellanox Technologies Ltd. 2001-2016.  ALL RIGHTS RESERVED.
+* See file LICENSE for terms.
+*/
+
+#ifndef UCT_SELF_EP_H
+#define UCT_SELF_EP_H
+
+#include <uct/base/uct_iface.h>
+
+
+typedef struct uct_self_ep {
```

Comment:
we probably don't need to add uct_self_ep, because it has no special data fields. we can just use uct_base_ep_t.